### ISSUE TITLE
wrong recurrence expand #3

### ISSUE THREAD

**Issue by petropavel13** [libical/libical#181]
I'm trying to expand following RRULE:
FREQ=DAILY;UNTIL=20150430T000000Z;BYDAY=SA,SU;BYMONTH=1,2,3,4,10,11,12
with DTSTART:
20141209T000000Z

```
ICAL_VERSION: 1.00
RRULE is: FREQ=DAILY;UNTIL=20150430T000000Z;BYDAY=SA,SU;BYMONTH=1,2,3,4,10,11,12
DTSTART is: 20141209T000000Z
icaltime is: 20141209T000000Z
icaltime is: 20141213T000000Z <- look here
icaltime is: 20141214T000000Z
icaltime is: 20141220T000000Z
icaltime is: 20141221T000000Z
icaltime is: 20141227T000000Z
icaltime is: 20141228T000000Z
icaltime is: 20141213T000000Z <- and then here
icaltime is: 20141214T000000Z
icaltime is: 20141220T000000Z
icaltime is: 20141221T000000Z
icaltime is: 20141227T000000Z
icaltime is: 20141228T000000Z
icaltime is: 20150103T000000Z <- repeats until end of month
...
```

Example code:
```c
#include <stdio.h>
#include <libical/ical.h>

int main(void)
{
    struct icalrecurrencetype recurrence = icalrecurrencetype_from_string("FREQ=DAILY;UNTIL=20150430T000000Z;BYDAY=SA,SU;BYMONTH=1,2,3,4,10,11,12");
    printf("ICAL_VERSION: %s\n", ICAL_VERSION);
    printf("RRULE is: %s\n", icalrecurrencetype_as_string(&recurrence));

    struct icaltimetype dtstart = icaltime_from_string("20141209T000000Z");
    printf("DTSTART is: %s\n", icaltime_as_ical_string(dtstart));

    icalrecur_iterator* const recur_iterator = icalrecur_iterator_new(recurrence, dtstart);

    struct icaltimetype ical_time = icalrecur_iterator_next(recur_iterator);

    while (!icaltime_is_null_time(ical_time)) {
        printf("icaltime is: %s\n", icaltime_as_ical_string(ical_time));
        ical_time = icalrecur_iterator_next(recur_iterator);
    }

    icalrecur_iterator_free(recur_iterator);

    return 0;
}
```
Can't find workaround for 1.0 version. Can you help me?

**Comment by ksmurchison** [comment]
This works correctly in master, I haven't tried with 1.0.1, but I believe this was fixed post-1.0.1

ICAL_VERSION: 2.0
RRULE is: FREQ=DAILY;UNTIL=20150430T000000Z;BYDAY=SA,SU;BYMONTH=1,2,3,4,10,11,12
DTSTART is: 20141209T000000Z
icaltime is: 20141213T000000Z
icaltime is: 20141214T000000Z
icaltime is: 20141220T000000Z
icaltime is: 20141221T000000Z
icaltime is: 20141227T000000Z
icaltime is: 20141228T000000Z
icaltime is: 20150103T000000Z
icaltime is: 20150104T000000Z
icaltime is: 20150110T000000Z
icaltime is: 20150111T000000Z
icaltime is: 20150117T000000Z
icaltime is: 20150118T000000Z
icaltime is: 20150124T000000Z
icaltime is: 20150125T000000Z
icaltime is: 20150131T000000Z
icaltime is: 20150201T000000Z
icaltime is: 20150207T000000Z
icaltime is: 20150208T000000Z
icaltime is: 20150214T000000Z
icaltime is: 20150215T000000Z
icaltime is: 20150221T000000Z
icaltime is: 20150222T000000Z
icaltime is: 20150228T000000Z
icaltime is: 20150301T000000Z
icaltime is: 20150307T000000Z
icaltime is: 20150308T000000Z
icaltime is: 20150314T000000Z
icaltime is: 20150315T000000Z
icaltime is: 20150321T000000Z
icaltime is: 20150322T000000Z
icaltime is: 20150328T000000Z
icaltime is: 20150329T000000Z
icaltime is: 20150404T000000Z
icaltime is: 20150405T000000Z
icaltime is: 20150411T000000Z
icaltime is: 20150412T000000Z
icaltime is: 20150418T000000Z
icaltime is: 20150419T000000Z
icaltime is: 20150425T000000Z
icaltime is: 20150426T000000Z

**Comment by winterz** [comment]
closing then
